apps/webapp/app/routes/_app.orgs.$organizationSlug.projects.v3.$projectParam.deployments/route.tsx

```diff
@@ -253,24 +269,51 @@ function DeploymentActionsCell({
   const location = useLocation();
   const project = useProject();
 
-  if (deployment.isCurrent || !deployment.isDeployed) return <TableCell to={path}>{""}</TableCell>;
+  const menuItems: JSX.Element[] = [];
+
+  if (!deployment.isCurrent && deployment.isDeployed) {
+    menuItems.push(
+      <Dialog>
+        <DialogTrigger asChild>
+          <Button variant="small-menu-item" LeadingIcon={ArrowUturnLeftIcon}>
+            Rollback
+          </Button>
+        </DialogTrigger>
+        <RollbackDeploymentDialog
+          projectId={project.id}
+          deploymentShortCode={deployment.shortCode}
+          redirectPath={`${location.pathname}${location.search}`}
+        />
+      </Dialog>
+    );
+  }
+
+  if (deployment.isLatest && deploymentIndexingIsRetryable(deployment)) {
+    menuItems.push(
+      <Dialog>
+        <DialogTrigger asChild>
+          <Button variant="small-menu-item" LeadingIcon={ArrowPathIcon}>
+            Retry indexing
+          </Button>
+        </DialogTrigger>
+        <RetryDeploymentIndexingDialog
+          projectId={project.id}
+          deploymentShortCode={deployment.shortCode}
+          redirectPath={`${location.pathname}${location.search}`}
+        />
+      </Dialog>
+    );
```

Comment:
Usually you would declare two booleans for this. and then use conditionals in the JSX

Reply:
Amateur hour..